crates/bevy_core_pipeline/src/taa/taa.wgsl

```diff
@@ -81,50 +74,45 @@ fn taa(@location(0) uv: vec2<f32>) -> Output {
     let texture_size = vec2<f32>(textureDimensions(view_target));
     let texel_size = 1.0 / texture_size;
 
-    // Fetch the current sample
-    let original_color = textureSample(view_target, nearest_sampler, uv);
-    var current_color = original_color.rgb;
-#ifdef TONEMAP
-    current_color = tonemap(current_color);
-#endif
-
-#ifndef RESET
-    // Pick the closest motion_vector from 5 samples (reduces aliasing on the edges of moving entities)
-    // https://advances.realtimerendering.com/s2014/index.html#_HIGH-QUALITY_TEMPORAL_SUPERSAMPLING, slide 27
-    let offset = texel_size * 2.0;
-    let d_uv_tl = uv + vec2(-offset.x, offset.y);
-    let d_uv_tr = uv + vec2(offset.x, offset.y);
-    let d_uv_bl = uv + vec2(-offset.x, -offset.y);
-    let d_uv_br = uv + vec2(offset.x, -offset.y);
+    // Loop over 3x3 neighborhood of the pre-TAA rendered texture
+    // https://alextardif.com/TAA.html
+    var current_color = vec3(0.0);
+    var weight_sum = 0.0;
+    var moment_1 = vec3(0.0);
+    var moment_2 = vec3(0.0);
+    var closest_depth = 0.0;
     var closest_uv = uv;
-    let d_tl = textureSample(depth, nearest_sampler, d_uv_tl);
-    let d_tr = textureSample(depth, nearest_sampler, d_uv_tr);
-    var closest_depth = textureSample(depth, nearest_sampler, uv);
-    let d_bl = textureSample(depth, nearest_sampler, d_uv_bl);
-    let d_br = textureSample(depth, nearest_sampler, d_uv_br);
-    if d_tl > closest_depth {
-        closest_uv = d_uv_tl;
-        closest_depth = d_tl;
-    }
-    if d_tr > closest_depth {
-        closest_uv = d_uv_tr;
-        closest_depth = d_tr;
-    }
-    if d_bl > closest_depth {
-        closest_uv = d_uv_bl;
-        closest_depth = d_bl;
-    }
-    if d_br > closest_depth {
-        closest_uv = d_uv_br;
+    var weights = array(1.5894572e-7, 0.8888889, 1.5894572e-7);
```

Comment:
are these weights correct? the outer weights are vanishingly small, the middle sample contributes 99.99993% of the total.

Reply:
Should be. It's this filter https://en.wikipedia.org/wiki/Mitchell%E2%80%93Netravali_filters, where B and C = 1/3, and x = sample distance from the center (0 or 1). That's the filter in 1d, and then to do it in 2d you just do MN(x) * MN(y).

So the two possible values are either:
1. `k(x = 0)` where B = C = 1/3
2. `k(x = 1)` where B = C = 1/3


And then each of 9 pixels you sample is one of those values times the other (represented by the weights array).

Reply:
ok but if the other 8 samples contribute 0.00007% to the final color, you could just not include them at all. and then you don't need the weights at all (or the div by weight_sum)

Reply:
It's supposed to be doing something. Try quickly removing it and compare screenshots. It should look slightly sharper with it.

Reply:
We could also try blackman-harris https://en.wikipedia.org/wiki/Window_function#Blackman%E2%80%93Harris_window

Reply:
i see no difference. where did you get the weights from? i don't see the numbers on the wiki page, and there's just no way such small weights will ever do anything. the edges are very close to f32::EPSILON and the corners are even smaller.

Reply:
The weights are `[k(1), k(0), k(1)]` from the wikipedia formula, where B = C = 1/3. I thought I tripled checked the math, but maybe I calculated them wrong? Or maybe we shouldn't use 1/3.

Reply:
Looking at the graph it could be correct, just not very useful with only 3 samples if you’re hitting the points which are zero on each wing?

Reply:
after reading the wiki page with a clearer head, it looks like this class of filter is meant for bilinear averaging over 4 pixels when the target point is not exactly on a pixel (ie working out how to blend nicely when you're zooming in to an image). so it makes sense that the weights for fully 1 pixel away from your sample origin are zero, and this doesn't look like the right type of filter to use here. 

i think using the middle colour directly is probably fine, and just using the surrounding pixles for the variance clipping.

Reply:
Unreal, Tomasz Stachowiak, and others have all mentioned that they use a filter like this to improve sharpness. I haven't been able to figure out how to do it though, so yeah I may just remove it from this PR.

Reply:
From a quick read of the karis slides, I *think* the idea is not to sample the neighbourhood but to weight the midpoint by the jitter offset. So rather than adding 100% weight * current frame to the history you add netravali-weight * current frame, and also maintain an exponentially decaying total weight to divide through by. 

I’m just guessing though, there must be a talk to go with the slides but I haven’t seen it

Reply:
I'm not sure, that's not what https://alextardif.com/TAA.html is doing. Search for `subSampleWeight`. I realized my code in this PR was broken, but "fixing" it to match what they did didn't actually do anything still.

Reply:
yes you're right ... in the link from there https://www.elopezr.com/temporal-aa-and-the-quest-for-the-holy-trail/ under "blurring" they have the filter being used to bilinearly filter the history, because the history for pixel minus motion vector doesn't lead to a precise pixel in the history. if that's what they are actually talking about then the b=1/3 c=1/3 netravali weights would be calculated and used instead of the w0/w12/w3 @ 127. then that makes total sense as it's a bilinear filter over the 4 nearest points. but maybe the cod4 cross filter is actually more recent / better than that.

also note the graph in the same section there shows ~0.056 for +/-1 (and that looks similar to the wiki graph as well) so it does look like there's something wrong with your weights...

Reply:
Ah so that's different, that's about sampling the _history_. I use a Catmull-Rom filter for that. The sharpness filter (M-N, or Blackman-Harris) is used for sampling the _view target_, i.e. the input to TAA.

Reply:
ok so the netravali sum reduces down to 
0: 1/6 * (6-2b) = 8/9 = 0.88889
+/-1: 1/6 * b = 1/18 = 0.05556

perhaps try that - we will have a real contribution from the neighbouring pixels.

Reply:
Did you account for the fact that the filter definitions operate on [-2, 2], while our sampling pattern is [-1, 1]? https://github.com/TheRealMJP/MSAAFilter/blob/master/MSAAFilter/Resolve.hlsl#L153

When I did the (correct) math, I still got basically-0 weights for the neighbors...

Reply:
In the formula for x>=1 it uses x^3, x^2, x and 1 factors, all of which boil down to 1 at x=1. So you can just sum all the bs and cs. The cs cancel out and you are left with +1b and the 1/6 outer term

Reply:
I'd still like confirmation that you accounted for the fact that the MV filter formula on wikipedia is [-2, 2] and not [-1, 1] (you need to rescale the input). But I used the coefficients you gave 3 posts above, and it seems to be working now :)

Box filter 
![image](https://github.com/bevyengine/bevy/assets/47158642/549697dd-2519-49f5-804d-f04d17bf18c7)
MV filter
![image](https://github.com/bevyengine/bevy/assets/47158642/a8b8eba5-1021-4595-b3b9-e8476baa0f44)

Reply:
sorry i missed that - i didn't use +/-2 as that yields exactly zero for the edges. i don't think the reference could be using +/-2 for that reason. note we are not operating on a 2x2 grid here like in a normal bilinear filter, but a 3x3 grid instead.

i just noticed alextardif was using `length(x,y)` rather than multiplying the axis terms - that would yield -0.0362113 for the corners instead of the current +0.003 (from 0.0556^2). might make it even sharper.

(if it should actually be using +/-2 offsets in each axis it would also need to take values outside the +/-2 range.. then it would be +0.8889 for the middle, 0 for the edges and -0.45 for each corner. but i really don't think it should be rescaled like this)

Reply:
Adjusting the corners made no difference. I think this PR is ready to go now.